src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -1012,6 +1013,31 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
         self ! InitialStartCall(callKey, CallActor.Start)
     }
 
+    /**
+      * If using Mac OS X with Docker Machine, by default Cromwell can only use the -v flag to 'docker run'
+      * if running from somewhere within the user's home directory.  If not, then -v will mount to the
+      * virtual machine that Docker Machine creates.  This ends up bubbling up to Cromwell when it can't find
+      * an RC file in the call's output directory (because it's actually on the virtual machine).
+      *
+      * This function will at least detect this case and log some WARN messages.
+      */
+    def detectDockerMachinePossibleMisusage(backendCall: LocalFileSystemBackendCall, logger: WorkflowLogger) = {
+      val os = Option(System.getProperty("os.name"))
+      val homeDir = Option(System.getProperty("user.home")).map(Paths.get(_).toAbsolutePath.toString)
+      val callRoot = backendCall.callRootPath.toAbsolutePath.toString
+      val dockerMachineName = sys.env.get("DOCKER_MACHINE_NAME")
+      if (dockerMachineName.nonEmpty && os.contains("Mac OS X") && homeDir.isDefined && !callRoot.startsWith(homeDir.get)) {
+        logger.warn(s"You are running Docker using Docker Machine and your working directory is not a subdirectory of ${homeDir.get}")
+        logger.warn(s"By default, Docker Machine only shares your home directory, which means Cromwell can only be run from under your home directory")
+        logger.warn(s"See https://docs.docker.com/engine/userguide/dockervolumes/ for more information")
+      }
+    }
+
+    backendCall match {
```

Comment:
Would that imply that you'll get this warning message even if you're not about to use the local backend?  What happens if you don't even have dockermachine/virtualbox installed?

Would it be possible for the backend interface to have a startup hook on it that gets invoked during server startup?  Then various checks, one time initialization, etc could happen there?  In the future there could be other lifecycle events as well.